test/Mono.Linker.Tests.Cases/DataFlow/PropertyDataFlow.cs

```diff
@@ -265,6 +267,27 @@ public void TestPropertyWithExistingAttributes ()
 				[param: DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicConstructors)]
 				set { PropertyWithExistingAttributes_Field = value; }
 			}
+
+			[UnrecognizedReflectionAccessPattern (typeof (PropertyWithIndexer), "Item.set", new Type[] { typeof (Type) }, messageCode: "IL2067")]
+			[LogDoesNotContain ("'Value passed to parameter 'index' of method 'Mono.Linker.Tests.Cases.DataFlow.PropertyDataFlow.TestAutomaticPropagationType.PropertyWithIndexer.Item.set'")]
+			public void TestPropertyWithIndexer (Type myType)
+			{
+				var propclass = new PropertyWithIndexer ();
+				propclass[1] = myType;
+				propclass[1].RequiresPublicConstructors ();
+			}
+
+			public class PropertyWithIndexer
+			{
+				[CompilerGenerated]
+				Type[] Property_Field;
```

Comment:
Nit: Just curious - I don't think the test needs the field to be compiler generated does it? It would work equally well if it was 
correctly annotated (just like the property).